src/ar_storage.erl

```diff
@@ -115,12 +118,26 @@ write_block(RawB) ->
 	end.
 -endif.
 
-%% Write a full block to disk, including writing TXs and modifying the
-%% TX list.
 write_full_block(B) ->
 	BShadow = B#block { txs = [T#tx.id || T <- B#block.txs] },
-	ar_storage:write_tx(B#block.txs),
-	ar_storage:write_block(BShadow).
+	write_full_block(BShadow, B#block.txs).
+
+write_full_block(BShadow, TXs) ->
+	%% We only store data that passes the firewall configured by the miner.
+	ScannedTXs = lists:filter(
+		fun(TX) ->
+			case ar_firewall:scan_tx(TX) of
+				accept ->
+					true;
+				reject ->
+					false
+			end
+		end,
+		TXs
+	),
+	write_tx(ScannedTXs),
+	write_block(BShadow),
+	ar_tx_search:update_tag_table(BShadow).
```

Comment:
We're passing in the non-filtered TXs to `ar_tx_search:update_tag_table/1` here. Shouldn't we do this instead:
`ar_tx_search:update_tag_table(BShadow#block { txs = ScannedTXs }).`

Reply:
Very good point!

Reply:
✅